### PR TITLE
refactor: Kotlin Cleanup: Collection/Note

### PR DESCRIPTION
**Pull request by david-allison**
## Checklist
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/libanki/Collection.kt [hunk 1]
```diff
@@ -1342,37 +1252,32 @@ open class Collection(
...
-    }
-
-    fun findDupes(fieldName: String?): List<Pair<String, List<Long>>> {
-        return Finder.findDupes(this, fieldName, "")
+        return Finder.findReplace(this, nids, src, dst, regex, field, fold)
     }
 
     @KotlinCleanup("JvmOverloads")
-    fun findDupes(fieldName: String?, search: String?): List<Pair<String, List<Long>>> {
+    fun findDupes(fieldName: String?, search: String? = ""): List<Pair<String, List<Long>>> {
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Collection.kt:1280]: This method is used only from kotlin code so the above @JvmOverloads cleanup is not needed anymore.

File: AnkiDroid/src/main/java/com/ichi2/libanki/Collection.kt [hunk 2]
```diff
@@ -1342,37 +1252,32 @@ open class Collection(
...
+        return Finder.findReplace(this, nids, src, dst, field = field)
     }
 
     @KotlinCleanup("JvmOverloads")
-    @KotlinCleanup("make non-null")
     fun findReplace(
-        nids: List<Long?>?,
-        src: String?,
-        dst: String?,
+        nids: List<Long?>,
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Collection.kt:1269]: This method is unused so I would add a cleanup with @KotlinCleanup("see if method can be removed").
Also, I would remove the cleanup for jvm overloads. While Finder.findReplace has default parameters and we could make the findReplace methods here with default parameter I wouldn't do it as it ties the two classes together(a change in Finder.findReplace default values would need to be replicated in Collection.findReplace).

**Review by lukstbit — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/main/java/com/ichi2/libanki/Collection.kt (hunk 2 above)
Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/libanki/Collection.kt:1268]: Done

I believe this is functionality which Anki Desktop exposes, and we'll want. to expose this to advanced users in the future.

**Review by lukstbit — APPROVED** [review]

**Review by mikehardy — APPROVED** [review]
:broom: :broom:

**Pull request merged**